strawberry/printer.py

```diff
@@ -1,7 +1,215 @@
+        return ""
+
+    prefer_multiple_lines = len(description) > 70
+    block_string = print_block_string(description, "", prefer_multiple_lines)
+
+    prefix = "\n" + indentation if indentation and not first_in_block else indentation
+
+    return prefix + block_string.replace("\n", "\n" + indentation) + "\n"
+
+
+def print_implemented_interfaces(
+    type_: Union[GraphQLObjectType, GraphQLInterfaceType]
+) -> str:
+    interfaces = type_.interfaces
+    return " implements " + " & ".join(i.name for i in interfaces) if interfaces else ""
+
+
+def print_input_value(name: str, arg: GraphQLArgument) -> str:
+    default_ast = ast_from_value(arg.default_value, arg.type)
+    arg_decl = f"{name}: {arg.type}"
+    if default_ast:
+        arg_decl += f" = {print_ast(default_ast)}"
+    return arg_decl
+
+
+def print_args(args: Dict[str, GraphQLArgument], indentation="") -> str:
+    if not args:
+        return ""
+
+    # If every arg does not have a description, print them on one line.
+    if not any(arg.description for arg in args.values()):
+        return (
+            "("
+            + ", ".join(print_input_value(name, arg) for name, arg in args.items())
+            + ")"
+        )
+
+    return (
+        "(\n"
+        + "\n".join(
+            print_description(arg, f"  {indentation}", not i)
+            + f"  {indentation}"
+            + print_input_value(name, arg)
+            for i, (name, arg) in enumerate(args.items())
+        )
+        + f"\n{indentation})"
+    )
+
+
+def print_deprecated(field_or_enum_value: Union[GraphQLField, GraphQLEnumValue]) -> str:
+    if not field_or_enum_value.is_deprecated:
+        return ""
+    reason = field_or_enum_value.deprecation_reason
+    reason_ast = ast_from_value(reason, GraphQLString)
+    if not reason_ast or reason == DEFAULT_DEPRECATION_REASON:
+        return " @deprecated"
+    return f" @deprecated(reason: {print_ast(reason_ast)})"
+
+
+def print_block(items: List[str]) -> str:
+    return " {\n" + "\n".join(items) + "\n}" if items else ""
+
+
+def print_federation_field_directive(field, metadata):
+    out = ""
+
+    if metadata and "federation" in metadata:
+        federation = metadata["federation"]
+
+        provides = federation.get("provides", "")
+        requires = federation.get("requires", "")
+        external = federation.get("external", False)
+
+        if provides:
+            out += f' @provides(fields: "{provides}")'
+
+        if requires:
+            out += f' @requires(fields: "{requires}")'
+
+        if external:
+            out += f" @external"
+
+    return out
+
+
+def print_fields(strawberry_type) -> str:
+    strawberry_fields = dataclasses.fields(strawberry_type)
+
+    def _get_metadata(field_name):
+        return next(
+            (
+                f.metadata
+                for f in strawberry_fields
+                if (getattr(f, "field_name", None) or f.name) == field_name
+            ),
+            None,
+        )
```

Comment:
we need this to map strawberry fields to graphql fields. Didn't find a way that didn't involve changing the whole print again (which we might want to do in future)